templates/routing-legacy/helper.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"github.com/nextmv-io/sdk/measure"
+	"github.com/nextmv-io/sdk/nextroute"
+	"github.com/nextmv-io/sdk/nextroute/common"
+	"github.com/nextmv-io/sdk/nextroute/schema"
+	"github.com/nextmv-io/sdk/route"
+	"github.com/twpayne/go-polyline"
+)
+
+func makeDistanceExpression(input schema.FleetInput) nextroute.DistanceExpression {
+	points := make([]measure.Point, len(input.Stops)+2*len(input.Vehicles))
+	for i, s := range input.Stops {
+		points[i] = measure.Point{
+			s.Position.Lon, s.Position.Lat,
+		}
+	}
+
+	counter := 0
+	for i := len(input.Stops); i < len(input.Stops); i += 2 {
+		if input.Vehicles[counter].Start != nil {
+			points[i] = measure.Point{
+				input.Vehicles[counter].Start.Lon, input.Vehicles[counter].Start.Lat,
+			}
+		}
+
+		if input.Vehicles[counter].End != nil {
+			points[i+1] = measure.Point{
+				input.Vehicles[counter].End.Lon, input.Vehicles[counter].End.Lat,
+			}
+		}
+		counter++
+	}
+
+	m := measure.HaversineByPoint()
+	mIndexed := measure.Indexed(m, points)
+
+	distanceMatrix := makeFloatMatrix(mIndexed, len(points))
+	distanceExpression := distanceExpression(&distanceMatrix)
+	return distanceExpression
+}
+
+// distanceExpression creates a distance expression for later use.
+func distanceExpression(distanceMatrix *[][]float64) nextroute.DistanceExpression {
+	distanceExpression := nextroute.NewHaversineExpression()
+	if distanceMatrix != nil {
+		distanceExpression = nextroute.NewDistanceExpression(
+			"travelDistance",
+			nextroute.NewMeasureByIndexExpression(measure.Matrix(*distanceMatrix)),
+			common.Meters,
+		)
+	}
+	return distanceExpression
+}
+
+type distanceData struct {
+	distance nextroute.DistanceExpression
+}
+
+// makeFloatMatrix is a helper function that takes a route.ByIndex and returns a
+// [][]float64.
+func makeFloatMatrix(matrix route.ByIndex, length int) [][]float64 {
```

Comment:
newFloatMatric, bit of nit picking but same argument as before use new pre-fix for factories